snapcraft/tests/test_base_plugin.py

```diff
@@ -172,34 +172,52 @@ class GetSourceTestCase(tests.TestCase):
             'source_type': 'bzr',
             'source_branch': 'test_branch',
             'source_tag': None,
+            'source_commit': None,
             'error': 'source-branch'}),
         ('tar with source branch', {
             'source_type': 'tar',
             'source_branch': 'test_branch',
             'source_tag': None,
+            'source_commit': None,
             'error': 'source-branch'}),
         ('tar with source tag', {
             'source_type': 'tar',
             'source_branch': None,
             'source_tag': 'test_tag',
+            'source_commit': None,
             'error': 'source-tag'}),
+        ('tar with source commit', {
+            'source_type': 'tar',
+            'source_branch': None,
+            'source_tag': None,
+            'source_commit': 'commit',
+            'error': 'source-commit'}),
         ('deb with source branch', {
             'source_type': 'deb',
             'source_branch': 'test_branch',
             'source_tag': None,
+            'source_commit': None,
             'error': 'source-branch'}),
         ('deb with source tag', {
             'source_type': 'deb',
             'source_branch': None,
             'source_tag': 'test_tag',
-            'error': 'source-tag'})
+            'source_commit': None,
+            'error': 'source-tag'}),
+        ('deb with source commit', {
+            'source_type': 'deb',
+            'source_branch': None,
+            'source_tag': None,
+            'source_commit': 'commit',
+            'error': 'source-commit'})
     ]
 
     @unittest.mock.patch('snapcraft.internal.sources._check_for_package')
     def test_get_source_with_branch_must_raise_error(self, mock_check):
         mock_check.side_effect = None
         options = tests.MockOptions('lp:this', self.source_type,
-                                    self.source_branch, self.source_tag)
+                                    self.source_branch, self.source_tag,
+                                    source_commit=self.source_commit)
```

Comment:
The inconsistency here is bothering me. I suggest we either make all the arguments positional or keyword.